safe-global/safe-wallet-web
# Refactor: Reuse Single account item component [SW-647]

## What it solves

Part of [SW-647](https://www.notion.so/safe-global/Refactor-Sidebar-components-1548180fe57380d08e77d05a3a121aec)
Resolves #4635 

## How this PR fixes it

- Removes `SubAccountItem` and instead reuses `SingleAccountItem` with an additional `isMultiChainItem` prop
- Also fixes the layout issue for the inconsistent signer warning on the dashboard
- Extracts `MultiAccountItem` logic into hooks
- Only mounts the `SafesList` under `MyAccounts` if the list is toggled
- -> This fixes the initial sidebar performance issue and moves the problem further down. To fully solve it we possibly need to use virtualized lists

## How to test it

1. Open the accounts list
2. Observe multichain items and single item still look the same as before

## Screenshots

<img width="802" alt="Screenshot 2024-12-13 at 13 53 49" src="https://github.com/user-attachments/assets/c10b8dab-87e0-4d55-9641-b2d13320a0bf" />
<img width="1285" alt="Screenshot 2024-12-13 at 13 54 12" src="https://github.com/user-attachments/assets/6840d1b6-9c3d-414c-b562-e179b38770a3" />


## Checklist
